runelite-client/src/main/java/net/runelite/client/plugins/prayerindicator/PrayerIndicatorPlugin.java

```diff
@@ -0,0 +1,172 @@
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.prayerindicator;
+
+import com.google.common.collect.MapDifference;
+import com.google.common.collect.Maps;
+import com.google.common.eventbus.Subscribe;
+
+import net.runelite.api.Client;
+import net.runelite.api.Prayer;
+import net.runelite.api.events.VarbitChanged;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.ui.overlay.infobox.InfoBoxManager;
+
+import javax.inject.Inject;
+import java.util.HashMap;
+import java.util.TreeMap;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+@PluginDescriptor(
+	name = "Prayer Indicator"
+)
+public class PrayerIndicatorPlugin extends Plugin
+{
+	private Map<Prayer, PrayerCounter> prayerOn = new TreeMap<>();
+	private Map<Prayer, PrayerCounter> prayerOnCache = new TreeMap<>();
+	private Map<Prayer, PrayerCounter> prayerOff = new TreeMap<>();
+	private Map<Prayer, PrayerCounter> prayerOffCache = new TreeMap<>();
+	private Map<Prayer, Integer> prayerMap = new HashMap<>();
+	private PrayerCounter[] prayerCounter = new PrayerCounter[PrayerType.values().length];
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private InfoBoxManager infoBoxManager;
+
+	@Override
+	protected void startUp()
+	{
+		ExecutorService executor = Executors.newFixedThreadPool(1);
+
+		Runnable runnable = () ->
+		{
+			for (PrayerType prayerType : PrayerType.values())
+			{
+				prayerCounter[prayerType.ordinal()] = new PrayerCounter(prayerType.getImage(), PrayerIndicatorPlugin.this, "", prayerType);
+			}
+		};
+
+		executor.execute(runnable);
+		executor.shutdown();
+	}
+
+	@Override
+	protected void shutDown()
+	{
+		infoBoxManager.removeIf(entry ->
```

Comment:
I dont think you want to remove all infoboxes on shutdown. `.removeIf(entry -> entry instanceOf PrayerCounter`

Reply:
Use the short notation. literally just write there what I wrote in the previous comment. `infoBoxManager.removeIf(entry -> entry instanceof PrayerCounter)` or even `infoBoxManager.removeIf(PrayerCounter::instanceOf)`